ocs_ci/deployment/aws.py

```diff
@@ -402,6 +422,386 @@ def deploy_ocp(self, log_cli_level='DEBUG'):
+            inst_id = response['Instances'][0]['InstanceId']
+            worker_ec2 = boto3.resource('ec2')
+            worker_instance = worker_ec2.Instance(inst_id)
+            worker_instance.wait_until_running()
+            worker_name = f'{cluster_id}-rhel-worker-{i}'
+            self.rhel_worker_list[worker_name] = worker_instance
+            worker_ec2.create_tags(
+                Resources=[inst_id],
+                Tags=[
+                    {'Key': 'Name', 'Value': f'{worker_name}'},
+                    {'Key': self.worker_tag[0], 'Value': self.worker_tag[1]}
+                ]
+            )
+            logging.info(self.worker_iam_role)
+            self.client.associate_iam_instance_profile(
+                IamInstanceProfile=self.worker_iam_role,
+                InstanceId=inst_id,
+            )
+
+    def run_ansible_playbook(self):
+        """
+        Bring up a helper pod (RHEL) to run openshift-ansible
+        playbook
+        """
+        rhel_pod_name = "rhel-ansible"
+        rhel_pod_obj = utils.create_rhelpod(
+            constants.DEFAULT_NAMESPACE, rhel_pod_name
+        )
+        timeout = 4000  # For ansible-playbook
+
+        # copy openshift-dev.pem to RHEL ansible pod
+        pem_src_path = "~/.ssh/openshift-dev.pem"
+        pem_dst_path = "/openshift-dev.pem"
+        pod.upload(rhel_pod_obj.name, pem_src_path, pem_dst_path)
+        repo_dst_path = "/etc/yum.repos.d/"
+        repo_file = os.path.basename(constants.OCP4_2_REPO)
+        pod.upload(
+            rhel_pod_obj.name, constants.OCP4_2_REPO, repo_dst_path
+        )
+        # copy the .pem file for our internal repo on all nodes
+        # including ansible pod
+        # get it from URL
+        mirror_pem_file_path = os.path.join(
+            constants.DATA_DIR,
+            constants.INTERNAL_MIRROR_PEM_FILE
+        )
+        dst = "/etc/pki/ca-trust/source/anchors/"
+        pod.upload(rhel_pod_obj.name, mirror_pem_file_path, dst)
+        # Install scp on pod
+        rhel_pod_obj.install_packages("openssh-clients")
+        # distribute repo file to all RHEL workers
+        hosts = [inst.private_dns_name for node, inst in
+                 self.rhel_worker_list.items()]
+        for host in hosts:
+            disable = "sudo yum-config-manager --disable *"
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path, disable, user=self.rhel_worker_user
+            )
+            rhel_pod_obj.copy_to_server(
+                host, pem_dst_path, f'{repo_dst_path}/{repo_file}',
+                f'/tmp/{repo_file}', user=self.rhel_worker_user
+            )
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path,
+                f'sudo mv /tmp/{repo_file} {repo_dst_path}',
+                user=self.rhel_worker_user
+            )
+            rhel_pod_obj.copy_to_server(
+                host, pem_dst_path,
+                f'{dst}/{constants.INTERNAL_MIRROR_PEM_FILE}',
```

Comment:
We should use `os.path.join()` instead of f-strings for building paths. With f-strings we might run into the issue of doubling `/`s which is the case here. `dst = "/etc/pki/ca-trust/source/anchors/"` so this f-string ends up being `"/etc/pki/ca-trust/source/anchors//ops-mirror.pem"`. This applies to all other instances of using f-strings for building paths in the PR.

Reply:
Done